posthog/api/person.py

```diff
@@ -98,15 +98,48 @@ def to_representation(self, instance: Person) -> Dict[str, Any]:
 
 class PersonFilter(filters.FilterSet):
     email = filters.CharFilter(field_name="properties__email")
-    distinct_id = filters.CharFilter(field_name="persondistinctid__distinct_id")
-    key_identifier = filters.CharFilter(method="key_identifier_filter")
+    distinct_id = filters.CharFilter(method="distinct_id_filter")
+    key_identifier = filters.CharFilter(method="key_identifier_filter", help_text="Filter on email or distinct ID")
+    uuid = filters.CharFilter(method="uuid_filter")
+    search = filters.CharFilter(method="search_filter")
+    cohort = filters.CharFilter(field_name="cohort__id", help_text="ID of a cohort the user belongs to")
+    properties = filters.CharFilter(method="properties_filter")
+
+    def __init__(self, data=None, queryset=None, *, request=None, prefix=None, team_id=None):
+        self.team_id = team_id
+        return super().__init__(data=data, queryset=queryset, request=request, prefix=prefix)
+
+    def distinct_id_filter(self, queryset, attr, *args, **kwargs):
+        queryset = queryset.filter(persondistinctid__distinct_id=args[0], persondistinctid__team_id=self.team_id)
```

Comment:
@hazzadous This is the speed up. Previously we didn't filter on the team_id which meant that we weren't hitting the index.

Reply:
Hm there's a super clear performance difference between these two queries:
```sql
 SELECT  "posthog_person"."id",
         "posthog_person"."created_at",
         "posthog_person"."properties_last_updated_at",
         "posthog_person"."properties_last_operation",
         "posthog_person"."team_id",
         "posthog_person"."properties",
         "posthog_person"."is_user_id",
         "posthog_person"."is_identified",
         "posthog_person"."uuid",
         "posthog_person"."version"
FROM "posthog_person"
INNER JOIN "posthog_persondistinctid"
    ON ("posthog_person"."id" = "posthog_persondistinctid"."person_id")
WHERE ("posthog_persondistinctid"."team_id" = 2
        AND "posthog_persondistinctid"."distinct_id" = '17ed94694a576-06365c84b9228d-43483550-32000-17ed94694a8a3')
ORDER BY  "posthog_person"."id" DESC LIMIT 100
```
is ~100ms vs
```sql
SELECT  "posthog_person"."id",
         "posthog_person"."created_at",
         "posthog_person"."properties_last_updated_at",
         "posthog_person"."properties_last_operation",
         "posthog_person"."team_id",
         "posthog_person"."properties",
         "posthog_person"."is_user_id",
         "posthog_person"."is_identified",
         "posthog_person"."uuid",
         "posthog_person"."version"
FROM "posthog_person"
INNER JOIN "posthog_persondistinctid"
    ON ("posthog_person"."id" = "posthog_persondistinctid"."person_id")
WHERE ("posthog_persondistinctid"."distinct_id" = '17ed94694a576-06365c84b9228d-43483550-32000-17ed94694a8a3')
ORDER BY  "posthog_person"."id" DESC LIMIT 100
```

which is 10 seconds!

Also explain shows that the first hits index, second does parallel seq scan.